bundles/org.eclipse.ui.workbench/eclipseui/org/eclipse/ui/internal/Workbench.java

```diff
@@ -820,8 +820,14 @@ public void run() throws Exception {
 
 					if (splashShell == null)
 						return;
-					if (background != null)
+					if (background != null) {
 						splashShell.setBackgroundImage(background);
+						Rectangle imageBounds = background.getBounds();
+						Rectangle shellBounds = splashShell.getBounds();
+						if (imageBounds.width > shellBounds.width || imageBounds.height > shellBounds.height) {
+							splashShell.setSize(imageBounds.width, imageBounds.height);
```

Comment:
Isn't the issue that the image is too large/small for the shell (because Platform UI applies quarter scaling while Equinox native launcher uses integer200) and not that the shell does not fit to the image? Changing the shell size here will result in a rescaling of the splash during application startup, won't it?